packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -321,11 +339,61 @@ class RenderGrid extends RenderBox with ContainerRenderObjectMixin<RenderBox, Gr
     assert(newDelegate != null);
     if (_delegate == newDelegate)
       return;
-    if (newDelegate.runtimeType != _delegate.runtimeType || newDelegate.shouldRelayout(_delegate))
+    if (newDelegate.runtimeType != _delegate.runtimeType || newDelegate.shouldRelayout(_delegate)) {
+      _specification = null;
       markNeedsLayout();
+    }
     _delegate = newDelegate;
   }
 
+  int get initialChildIndex => _initialChildIndex;
+  int _initialChildIndex;
+  void set initialChildIndex(int value) {
+    assert(value != null);
+    if (_initialChildIndex == value)
```

Comment:
> Not sure if you can do something like that in Dart.

Unfortunately, that pattern has a runtime cost in Dart, unlike in C++.  You're trading verbosity for polymophism, which in this case adds to megamorphic calls (the operator== and the callback).  In C++, you'd use a template, which would generate code at compile time to specialize the source code for each type of caller.  Dart doesn't have that sort of code generation, so we use verbosity instead.